src/lib.rs

```diff
@@ -1,63 +1,168 @@
+//!         geometry: Geometry {
+//!             value: Value::LineString(vec![0]),
+//!             bbox: None,
+//!             id: None,
+//!             properties: None,
+//!             foreign_members: None,
+//!         },
+//!     }],
+//!     bbox: None,
+//!     transform: None,
+//! };
+//!
+//! let topojson_string = serde_json::to_string(&topo).unwrap();
+//! ```
+//!
+//! ## Converting to GeoJson
+//!
+//!
+//! `TopoJson` can then be converted to `GeoJson using the `to_geojson` function:
+//!
+//! ```rust
+//! # extern crate topojson;
+//! extern crate geojson;
+//!
+//! use topojson::{TopoJson, to_geojson};
+//! use geojson::GeoJson;
+//!
+//! let topojson_str = r#"
+//! {
+//!     "arcs": [[[0.0, 0.0], [0.0, 9999.0], [2000.0, 0.0], [0.0, -9999.0], [-2000.0, 0.0]]],
+//!     "objects": {"example": { "type": "GeometryCollection", "geometries": [
+//!         {"coordinates": [4000.0, 5000.0], "properties": {"prop0": "value0"}, "type": "Point"},
+//!         {"arcs": [[0]], "properties": {"prop0": "value0", "prop1": {"this": "that"}}, "type": "Polygon"}]}},
+//!     "transform": {"scale": [0.0005, 0.0001], "translate": [100.0, 0.0]},
+//!     "type": "Topology"
+//! }"#;
+//! // Parse to Topology:
+//! let topo = topojson_str.parse::<TopoJson>().unwrap();
+//!
+//! // Conversion to GeoJson FeatureCollection for the "example" object:
+//! let geojson = match topo {
+//!     TopoJson::Topology(t) => {
+//!         to_geojson(&t, &String::from("example"))
+//!             .expect("Unable to convert TopoJSON to GeoJSON")
+//!     },
+//!     _ => unimplemented!(),
+//! };
+//! ```
+//!
+extern crate serde;
+#[macro_use]
+extern crate serde_json;
 
-macro_rules! expect_string {
-    ($value:expr) => (try!(
-        match $value.as_string() {
-            Some(v) => Ok(v),
-            None => Err({use TopoJsonError; TopoJsonError::new("Expected string value")})
-        }
-    ))
-}
+extern crate geojson;
+/// Bounding Boxes
+///
+/// [TopoJSON Format Specification § 3](https://github.com/topojson/topojson-specification#3-bounding-boxes)
+pub type Bbox = Vec<f64>;
 
-macro_rules! expect_i64 {
-    ($value:expr) => (try!(
-        match $value.as_i64() {
-            Some(v) => Ok(v),
-            None => Err({use TopoJsonError; TopoJsonError::new("Expected i64 value")})
-        }
-    ))
-}
+/// Positions
+///
+/// [TopoJSON Format Specification § 2.1.1](https://github.com/topojson/topojson-specification#211-positions)
+pub type Position = Vec<f64>;
+// pub type PointType = Position;
 
-macro_rules! expect_f64 {
-    ($value:expr) => (try!(
-        match $value.as_f64() {
-            Some(v) => Ok(v),
-            None => Err({use TopoJsonError; TopoJsonError::new("Expected f64 value")})
-        }
-    ))
-}
+/// Arcs (an array of position which may have been quantized and delta-encoded)
+///
+/// [TopoJSON Format Specification $ 2.1.3](https://github.com/topojson/topojson-specification#213-arcs)
+pub type Arc = Vec<Position>;
```

Comment:
It's a bit of a shame that `Arc` already has a meaning in Rust as _atomic reference counted_. Maybe we should note in the docs themselves that `Arc` means something else here in addition to the link to the spec.